collector/collector.go

```diff
@@ -78,11 +80,22 @@ func (c *collector) Collect(from, to time.Time) error {
 }
 
 func (c *collector) CollectTodaysMetrics() error {
+	todaysLatencies, err := c.Source.TodaysLatencies()
+	if err != nil {
+		return err
+	}
+	c.Logger.WithFields(logger.Fields{"todays_latencies_count": len(todaysLatencies)}).Info("Collected todays latencies")
+
 	todaysCounts, err := c.Source.TodaysCounts()
 	if err != nil {
 		return err
 	}
-	c.Logger.WithFields(logger.Fields{"todays_metrics_count": len(todaysCounts)}).Info("Collected todays metrics")
+	c.Logger.WithFields(logger.Fields{"todays_usage_count": len(todaysCounts)}).Info("Collected todays usage")
+
+	err = c.Writer.WriteTodaysLatencies(todaysLatencies)
+	if err != nil {
```

Comment:
Same as previous comment re errors encountered processing latency data: I don't think other metrics should be skipped. 

In addition, this function returns multiple, not necessarily related, errors: it likely needs a refactor: e.g. `WriteTodaysMetrics` as a single function?

Reply:
Updated